extension/dynamicconfig/factory.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamicconfig
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+const (
+	// The value of extension "type" in configuration.
+	typeStr = "dynamicconfig"
+)
+
+// Factory is the factory for the extension.
+type Factory struct{}
+
+// Type gets the type of the config created by this factory.
+func (f *Factory) Type() configmodels.Type {
+	return typeStr
+}
+
+// CreateDefaultConfig creates the default configuration for the extension.
+func (f *Factory) CreateDefaultConfig() configmodels.Extension {
+	return &Config{
+		ExtensionSettings: configmodels.ExtensionSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
+		Endpoint:            "0.0.0.0:55700",
+		RemoteConfigAddress: "",
+		LocalConfigFile:     "schedules.yaml",
```

Comment:
Should we change this to "local-schedules.ymal"? I don't find such a file other than in testdata. There is a local-schedules.yaml in the example directory though.

Reply:
Good thought, I switched to just "schedules.yaml" since it seemed a little more concise. "local-schedules.yaml" felt a little redundant, since if the schedules are hosted locally, then it's clear they're "local." I've gone ahead and renamed the file in the example directory to "schedules.yaml," if that sounds all right?